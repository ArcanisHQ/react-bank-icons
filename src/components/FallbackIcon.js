import React from 'react'

const FallbackIcon = () => {
    return (
        <div>
            Error.
        </div>
    )
}
export default FallbackIcon;
