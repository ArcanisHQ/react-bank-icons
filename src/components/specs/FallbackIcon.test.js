jest.mock('../../icons');
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import FallbackIcon from '../FallbackIcon';

const defaultProps = {
  color: 'white',
  size: 16,
};

const BankIconComponent = (props = defaultProps) => shallow(<FallbackIcon {...props} />);
const findChild = (selector) => BankIconComponent().find(selector);

describe('<FallbackIcon />', () => {
  it('renders BankIcon component correctly', () => {
    const component = renderer.create(<FallbackIcon {...defaultProps} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have an svg', () => {
    expect(findChild('svg')).toBeTruthy();
  });
});
