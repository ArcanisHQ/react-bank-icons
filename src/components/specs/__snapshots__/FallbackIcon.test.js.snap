// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FallbackIcon /> renders BankIcon component correctly 1`] = `
<svg
  fill="white"
  height={16}
  viewBox="0 0 24 24"
  width={16}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Banco
  </title>
  <desc>
    Ã­cone representando um banco.
  </desc>
  <path
    clipRule="evenodd"
    d="M3 0C1.34315 0 0 1.34315 0 3V21C0 22.6569 1.34315 24 3 24H21C22.6569 24 24 22.6569 24 21V3C24 1.34315 22.6569 0 21 0H3ZM20.7115 9.17332C21.2572 9.57118 20.9767 10.4343 20.305 10.4343H18.957V18.5654H20.5224C20.618 18.5654 20.6963 18.6436 20.6963 18.7393V19.8698C20.6963 19.9655 20.618 20.0438 20.5224 20.0438H3.47753C3.38187 20.0438 3.3036 19.9655 3.3036 19.8698V18.7393C3.3036 18.6436 3.38187 18.5654 3.47753 18.5654H5.04287V10.4343H3.69711C3.02314 10.4343 2.74269 9.569 3.28838 9.17332L11.5912 3.13371C11.7098 3.04683 11.853 3 11.9999 3C12.1469 3 12.2901 3.04683 12.4087 3.13371L20.7115 9.17332ZM6.60821 18.5654H9.15189V10.4343H6.60821V18.5654ZM10.7172 18.5654H13.2609V10.4343H10.7172V18.5654ZM14.8263 18.5654H17.3917V10.4343H14.8263V18.5654Z"
    fillRule="evenodd"
  />
</svg>
`;
